remote-api/src/main/java/org/alfresco/rest/api/impl/CategoriesImpl.java

```diff
@@ -106,17 +105,36 @@ public List<Category> createSubcategories(String parentCategoryId, List<Category
                 .collect(Collectors.toList());
     }
 
-    private NodeRef createCategoryNodeRef(NodeRef parentNodeRef, Category c)
+    @Override
+    public Category updateCategoryById(final String id, final Category fixedCategoryModel)
     {
-        if (StringUtils.isEmpty(c.getName())) {
-            throw new InvalidArgumentException(NOT_NULL_OR_EMPTY);
+        verifyAdminAuthority();
+        final NodeRef categoryNodeRef = nodes.validateOrLookupNode(id, null);
+        if (isNotACategory(categoryNodeRef) || isRootCategory(categoryNodeRef))
+        {
+            throw new InvalidArgumentException(NOT_A_VALID_CATEGORY, new String[]{id});
         }
-        return categoryService.createCategory(parentNodeRef, c.getName());
+
+        final Category originalCategory = mapToCategory(categoryNodeRef);
+        compareAndVerifyCategoryFields(fixedCategoryModel, originalCategory);
+
+        return mapToCategory(changeCategoryName(categoryNodeRef, fixedCategoryModel.getName()));
     }
 
-    private boolean isNotACategory(NodeRef nodeRef)
+    private void verifyAdminAuthority()
     {
-        return !nodes.isSubClass(nodeRef, ContentModel.TYPE_CATEGORY, false);
+        if (!authorityService.hasAdminAuthority())
+        {
+            throw new PermissionDeniedException(NO_PERMISSION_TO_CREATE_A_CATEGORY);
+        }
+    }
+
+    private NodeRef createCategoryNodeRef(NodeRef parentNodeRef, Category c)
+    {
+        if (StringUtils.isEmpty(c.getName())) {
```

Comment:
\[Formatting\] Trailing braces a couple of times in this file.

Reply:
Oh - incidentally we can use org.alfresco.util.ParameterCheck.mandatoryString for this.